org.eclipse.january.examples/src/org/eclipse/january/examples/dataset/SlicingExamples.java

```diff
@@ -66,5 +71,26 @@ public void iterateImages2() throws DatasetException {
 		}
 	}
 
+	/**
+	 * Slice using basic Slice object
+	 * @throws Exception 
+	 */
+	@Test
+	public void iterateImagesND() throws DatasetException {
+		
+		final ILazyDataset lz = Random.lazyRand(64, 64, 100, 100);		
+		final PositionIterator it = new PositionIterator(new int[]{64, 64});
+		
+		while(it.hasNext()) {
+			int[] pos = it.getPos();
```

Comment:
The most efficient way is to use the full iterator but miss out the last two dimensions and then use a SliceND object
```
idim = 2; // dimension that image starts
it = new PositionIterator(lz.getShape(), 2, 3); // or ...,-2,-1
slicing = new Slice[idim]; // define a ND slice
for (int i = 0; i < idim; i++) {
    slicing[i] = new Slice(1);
}
slice = new SliceND(lz.getShape(), slicing);
start = slice.getStart();
stop = slice.getStop();

while (it.hasNext()) {
    for (int i = 0; i < idim; i++) {
        int p = pos[i];
        start[i] = p;
        stop[i] = p+1;
    }
    // ...
}